replicator.js

```diff
@@ -1,101 +1,213 @@
-// Define an array named `properties` with objects representing
-// the user-adjustable inputs to your application
-var properties = [
-  {type: 'range', id: 'Columns', value: 2, min: 1, max: 10, step: 1},
-  {type: 'range', id: 'Rows', value: 2, min: 1, max: 10, step: 1},
-  {type: 'range', id: 'Spacing (in)', value: 0.5, min: 0, max: 1, step: 0.001},
-];
-
-var shallowCopy = function(object) {
+var COLUMNS = "Columns";
+var ROWS = "Rows";
+var CENTERS = "Use center distance";
+var SPACING;
+
+function properties(args) {
+  SPACING = `Spacing (${args.preferredUnit})`;
+
+  return [
+    {
+      type: "range",
+      id: COLUMNS,
+      value: 2,
+      min: 1,
+      max: 10,
+      step: 1
+    },
+    {
+      type: "range",
+      id: ROWS,
+      value: 2,
+      min: 1,
+      max: 10,
+      step: 1
+    },
+    {
+      type: "range",
+      id: SPACING,
+      value: 0.5,
+      min: 0,
+      max: 1,
+      step: 0.001
```

Comment:
I thought the same thing when I first used the Replicator -- it seemed odd that it was a range control.

I'll change this to a text input, and convert it to numbers on use.

BTW, for context, I started this PR because I was using my custom XCarve to cut holes in its bed for threaded inserts, and wanted them to be 75mm on-center.